src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -616,57 +615,44 @@ class FrontendGenImpl {
         outputTypes.emplace_back(builder_.getNoneType());
 
     // TODO: Handle optional inputs.
-    auto op =
-        builder_.create<T>(ImportLoc(node), outputTypes, inputs, attributes);
-    Operation *genericOp = op.getOperation();
+    T op = builder_.create<T>(ImportLoc(node), outputTypes, inputs, attributes);
     // Type inference for results.
     for (const auto &attr : node.attribute()) {
       if (attr.type() == onnx::AttributeProto_AttributeType_GRAPH) {
-        if (auto opWithSubgraph =
-                dyn_cast<HasOnnxSubgraphOpInterface>(op.getOperation())) {
-          auto regionIdx = opWithSubgraph.getSubgraphRegionIdx(attr.name());
-          auto &region = op->getRegion(regionIdx);
-          region.push_back(new Block);
-          OpBuilder::InsertionGuard guard(builder_);
-          builder_.setInsertionPointToStart(&region.back());
-          auto funcType =
-              importGraph(attr.g(), region, op.getOperation(), false);
-          // Use type info from graph to reset type of output for current op
-          for (int i = 0; i < node.output().size(); i++) {
-            Type type = funcType.getResults()[i];
-            if ((num_use_inference_outputs < 0) ||
-                (i < num_use_inference_outputs)) {
-              genericOp->getOpResult(i).setType(type);
-            } else {
-              TensorType tensorType = type.cast<TensorType>();
-              Type extendedType =
-                  UnrankedTensorType::get(tensorType.getElementType());
-              genericOp->getOpResult(i).setType(extendedType);
-            }
```

Comment:
I suspect that this didn't work in all cases: ImportScan() called with num_use_inference_outputs set to the value of the "num_scan_inputs" attribute, but that counts trailing scan inputs while the logic here counts leading non-scan outputs